packages/angular/ssr/src/utils/crypto.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/**
+ * Generates a SHA-256 hash of the provided string.
+ *
+ * @param data - The input string to be hashed.
+ * @returns A promise that resolves to the SHA-256 hash of the input,
+ * represented as a hexadecimal string.
+ */
+export async function sha256(data: string): Promise<string> {
+  if (typeof crypto === 'undefined') {
+    // TODO(alanagius): remove once Node.js version 18 is no longer supported.
+    throw new Error(
```

Comment:
This code is non Node.js specific, but from the runtimes I checked I was only Node.js version 18 which doesn’t have a global web crypto, considering this version is in maintenance mode, and that it will be EOF in April 2025, I didn’t opt for doing a fallback. 

It is also worth noting that using Node.js 20 or later also provides better performance.

Reply:
Thanks for additional info 👍